secure/key-manager/src/lib.rs

```diff
@@ -123,13 +123,24 @@ where
 
     /// Begins execution of the key manager by running an infinite loop where the key manager will
     /// periodically wake up, verify the state of the validator keys (e.g., the consensus key), and
-    /// initiate a key rotation when required. If something goes wrong, an error will be returned
-    /// by this method, upon which the key manager will flag the error and stop execution.
+    /// initiate a key rotation when required. If something goes wrong that we can't handle, an
+    /// error will be returned by this method, upon which the key manager will flag the error and
+    /// stop execution.
     pub fn execute(&mut self) -> Result<(), Error> {
         info!("The key manager has been created and is starting execution.");
         loop {
             info!("Checking the status of the keys.");
-            self.execute_once()?;
+            match self.execute_once() {
+                Ok(_) => {} // Expected case
+                Err(Error::LivenessError(last_value, current_value)) => {
+                    // Log the liveness error, but don't throw the error up the call stack.
+                    error!(
+                        "Encountered error, but still continuing to execute: {}",
+                        Error::LivenessError(last_value, current_value).to_string()
+                    );
+                }
+                Err(e) => return Err(e), // Unexpected error that we can't handle -- throw!
```

Comment:
I think it is okay to infinite loop here. We should use logs and our kubernetes service to detect stalled processes.

Reply:
Okay, I'll address this later on when I come to handling docker and deployment of the key manager. The issue with making this change right now is that if we don't kill the key manager on an error, we can't call "key_manager.wait()" in the test, because the key manager will never terminate.. I'd like to have this smoke test run a little bit first in the dev environments, because I also am not sure how flaky it will be right now.. :(